packages/core/components/FileMetadataSearchBar/index.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+    DatePicker,
+    Dropdown,
+    Icon,
+    IconButton,
+    IDropdownOption,
+    SearchBox,
+} from "@fluentui/react";
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { AnnotationName, TOP_LEVEL_FILE_ANNOTATIONS } from "../../constants";
+import { AnnotationType } from "../../entity/AnnotationFormatter";
+import FileFilter from "../../entity/FileFilter";
+import { selection } from "../../state";
+import { getFileAttributeFilter } from "../../state/selection/selectors";
+
+const styles = require("./FileMetadataSearchBar.module.css");
+
+const FILE_ATTRIBUTE_OPTIONS = TOP_LEVEL_FILE_ANNOTATIONS.filter(
+    (a) => a.name !== AnnotationName.FILE_SIZE
+).map((a) => ({
+    key: a.name,
+    text: a.displayName,
+}));
+const FILE_NAME_OPTION = FILE_ATTRIBUTE_OPTIONS.find(
+    (o) => o.key === AnnotationName.FILE_NAME
+) as IDropdownOption;
+// Color chosen from App.module.css
+const PURPLE_ICON_STYLE = { icon: { color: "#827aa3" } };
+export const DATE_RANGE_SEPARATOR = "-to-"; // Not arbitrary, defined per contract with FES
+
+// Because the datestring comes in as an ISO formatted date like 01-02-21
+// creating a new date from that would result in a date displayed as the
+// day before due to the UTC offset, to account for this we can add in the offset
+// ahead of time.
+function extractDateFromDateString(dateString?: string): Date | undefined {
```

Comment:
Might as well write [a few](https://mochajs.org/#dynamically-generating-tests) test cases for this